### PR TITLE
Base pool.cpu_info:features_hvm on HVM-capable hosts only

### PR DESCRIPTION
**Pull request by robhoes**
This field should be the intersection of the HVM-specific feature sets over all
hosts in the pool that can actually run HVM guests.

The `host.cpu_info:features_hvm` field of non-HVM-capable hosts is a string
with just zeros: no features. Therefore, the intersection currently ends up
disabling all features for HVM guests in the entire pool (including for HVM
capable hosts), which is obviously not what we want.

Signed-off-by: Rob Hoes <rob.hoes@citrix.com>

### CODE REVIEW

**Comment by mseri** [comment]
This broke lots of tests: https://travis-ci.org/xapi-project/xen-api/jobs/380633964

**Comment by robhoes** [comment]
Unit test problem :(

**Comment by robhoes** [comment]
I've fixed the tests.

File: ocaml/xapi/helpers.ml [hunk 1]
```diff
@@ -1207,3 +1207,10 @@ let get_first_vusb ~__context usb_group =
     raise Api_errors.(Server_error(internal_error,
       [Printf.sprintf "there is no VUSB associated with the USB_group: %s"
       (Ref.string_of usb_group)]))
+
+let host_supports_hvm ~__context host =
+  (* We say that a host supports HVM if any of
+   * the capability strings contains the substring "hvm". *)
+  let capabilities = Db.Host.get_capabilities ~__context ~self:host in
+  List.fold_left (||) false
```
[the remark this answers is not in the document]
Comment (robhoes) [reply ocaml/xapi/helpers.ml]: Or does the `||` not evaluate the right side if the left is `true`?

[the remark this answers is not in the document]
Comment (robhoes) [reply ocaml/xapi/helpers.ml]: Yes, I tried, you're right :)

**Pull request merged**